# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/ForOfEmitter.cpp
# Commit: 7a3a4207c55b
# Full Hash: 7a3a4207c55b27d6a73783c0021cfed2368bdbb8
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-06 21:48:33
# Regressor Bug: 1601897
# File Overlap Count: 3
# Description:
#   Bug 1601897 part 2 - Move patchBreaks and addTryNote calls to LoopControl::emitLoopEnd. r=arai
#   
#   This ensures all loops handle this the same way.
#   
#   For JSTRY_FOR_IN some places asserted the try note ends at the
# ==============================================================================

diff -r 7a9ab8423e85 -r 7a3a4207c55b js/src/frontend/ForOfEmitter.cpp
--- a/js/src/frontend/ForOfEmitter.cpp	Fri Dec 06 11:08:02 2019 +0000
+++ b/js/src/frontend/ForOfEmitter.cpp	Fri Dec 06 11:08:04 2019 +0000
@@ -233,23 +233,13 @@
     }
   }
 
-  if (!loopInfo_->emitLoopEnd(bce_, JSOP_GOTO)) {
+  if (!loopInfo_->emitLoopEnd(bce_, JSOP_GOTO, JSTRY_FOR_OF)) {
     //              [stack] NEXT ITER UNDEF
     return false;
   }
 
   MOZ_ASSERT(bce_->bytecodeSection().stackDepth() == loopDepth_);
 
-  if (!loopInfo_->patchBreaks(bce_)) {
-    return false;
-  }
-
-  if (!bce_->addTryNote(JSTRY_FOR_OF, bce_->bytecodeSection().stackDepth(),
-                        loopInfo_->headOffset(),
-                        loopInfo_->breakTargetOffset())) {
-    return false;
-  }
-
   if (!bce_->emitPopN(3)) {
     //              [stack]
     return false;